src/nvme/mi.h

```diff
@@ -307,6 +310,229 @@ enum nvme_mi_config_smbus_freq {
 	NVME_MI_CONFIG_SMBUS_FREQ_1MHz = 0x3,
 };
 
+/* Asynchronous Event Message definitions*/
+
+/**
+ * struct nvme_mi_aem_supported_list_header - Asynchronous Event Supported List Header.
+ * @numaes: Number of AE supported data structures that follow the header
+ * @aeslver: AE Supported List Version
+ * @aest: AE Supported list length (including this header)
+ * @aeslhl: AE Supported list header length
+ *
+ * This header preceeds a number, (&numaes), of AE supported data structures
+ */
+struct nvme_mi_aem_supported_list_header {
+	__u8 numaes;
+	__u8 aeslver;//Should be zero
+	__le16 aest;
+	__u8 aeslhl; //Should be 5
+} __attribute__((packed));
+
+/**
+ * struct nvme_mi_aem_supported_item - AE Supported List Item
+ * @aesl: AE supported list item length
+ * @aesi: AE supported info
+ *
+ * Following this header should be hdr.numaes entries of
+ * nvme_mi_aem_supported_item structures
+ */
+struct nvme_mi_aem_supported_item {
+	__u8 aesl;//Length of this item.  Set to 3
+	__le16 aesi;
+} __attribute__((packed));
+
+/**
+ * nvme_mi_aem_aesi_get_aese() - return aese from aesi field
+ * @aesi: aesi field from @nvme_mi_aem_supported_item
+ *
+ * Returns: A bool representing the aese value
+ */
+bool nvme_mi_aem_aesi_get_aese(__le16 aesi);
+
+/**
+ * nvme_mi_aem_aesi_get_aesid() - return aesid from aesi field
+ * @aesi: aesi field from @nvme_mi_aem_supported_item
+ *
+ * Returns: aesid value
+ */
+__u8 nvme_mi_aem_aesi_get_aesid(__le16 aesi);
+
+/**
+ * nvme_mi_aem_aesi_set_aesid() - set aesid in the aesi field
+ * @aesid: aesid value to use
+ * @item: Pointer to @nvme_mi_aem_supported_item to update the aesi field
+ */
+void nvme_mi_aem_aesi_set_aesid(__u8 aesid, struct nvme_mi_aem_supported_item *item);
```

Comment:
I'd tend to have the modified thing (the "context") as the first argument.

(Also for the `_set` functions below too)